### PR TITLE
Fix form behavior error summary duplicates

### PR DESCRIPTION
**Pull request by homing1**
Task: task-[662148](https://dev.azure.com/ceapex/Engineering/_workitems/edit/662148/), task-[655260](https://dev.azure.com/ceapex/Engineering/_workitems/edit/655260/)

Link: preview-[421](https://design.docs.microsoft.com/pulls/421/components/form.html)

This PR aims to achieve the following:
- Remove duplicate messages in the form behavior error summary for Firefox.
- Adjust the scroll margin for field body content elements.
- Adds focus to star rating as custom elements do not have focus by default.

## Testing

1. Visit https://design.docs.microsoft.com/pulls/421/components/form.html with Firefox. Scroll down to form behavior section. Click on its Submit button.
2. The error summary should be visible and it should not contain duplicate messages. Check with other browsers.
image
3. Click on "A selection for "Star rating" is required" error message link.  The window should jump to the star rating and its label should be in view after the jump. The first star should have focus outline.
image

### CODE REVIEW

File: js/src/elements/form-behavior.ts [hunk 1]
```diff
@@ -709,9 +719,11 @@ export function navigateAfterSubmit(href: string, navigationStep: NavigationStep
 export function collectCustomElementsByName(form: HTMLFormElement): Element[] {
 	// Compare FormData with form.elements to find missing elements.
 	const formData = Object.fromEntries(new FormData(form));
-	const formElements = Object.keys(form.elements);
 	const customElementList: Element[] = [];
-	const customElements = Object.keys(formData).filter(el => formElements.indexOf(el) === -1);
+	const customElements = Object.keys(formData).filter(el => {
```
Comment (wibjorn) [inline js/src/elements/form-behavior.ts:723]: nit: `el` makes this seem like an element type, but real time is a string, should it be something like `key`, or `name` just to help with readability. Resolve as you please

File: js/src/elements/star-rating.ts [hunk 2]
```diff
@@ -314,6 +316,9 @@ export class StarRatingElement extends HTMLElement {
 				this.updateContent('value', target.value, target);
 				this.dispatchEvent(new Event('change', { bubbles: true }));
 				break;
+			case 'focus':
+				this.shadowRoot!.querySelectorAll('input')[0].focus();
```
Comment (wibjorn) [inline js/src/elements/star-rating.ts:320]: When I click on the fourth star, the first star gets a focus-visible outline, this is desireable?

File: js/src/elements/form-behavior.ts (hunk 1 above)
Comment (homing1) [reply js/src/elements/form-behavior.ts]: That's true, updated

File: js/src/elements/star-rating.ts (hunk 2 above)
Comment (homing1) [reply js/src/elements/star-rating.ts]: No that was an unintended side effect, should be fixed now.

**Comment by wibjorn** [comment]
Tiny nit: no need to hold up this PR, but the focus rect is a little wonky on start rating:

image

**Review by wibjorn — APPROVED** [review]

File: js/src/elements/form-behavior.ts [hunk 3]
```diff
@@ -266,7 +266,6 @@ class FormBehaviorElement extends HTMLElement {
 		errorList: HTMLUListElement;
 	} {
 		const formLayout = form.querySelector('.form-error-container') || form;
```
[the remark this answers is not in the document]
Comment (homing1) [reply js/src/elements/form-behavior.ts:268]: I'm actually going to be removing tags entirely in another PR, resolving.

**Comment by homing1** [comment]
> Tiny nit: no need to hold up this PR, but the focus rect is a little wonky on start rating:
> 
> image

I'll table this for later. It also doesn't look great on Firefox.

**Review by BenDMyers — APPROVED** [review]

**Pull request merged**